### PR TITLE
fix(local-rec) Handle no mic permission

### LINKED ISSUE #12857: Local Recording fails silently if microphone permission not granted
### Description:

If microphone permission is not granted, local recording appears to work when started but no recording file is generated when recording is stopped.

Related community post: https://community.jitsi.org/t/jitsi-local-recording-not-working/120476/14?u=shawn

### Steps to reproduce:

1. Join a meeting using a new private window in browser
2. Deny permissions for camera and microphone 
3. Start local recording
4. Stop local recording

### Expected behavior:

Recording file is offered up for download.

### Actual behavior:

No recording file available. Console log shows the following error:

```
Logger.js:154 2023-02-01T19:46:37.391Z [JitsiMeetJS.ts] <Object.getGlobalOnErrorHandler>:  
UnhandledError: Cannot read properties of undefined (reading 'dataEnd') Script: null Line: null 
Column: null StackTrace:  TypeError: Cannot read properties of undefined (reading 'dataEnd')
    at t.makeMetadataSeekable (https://meet.domain.com/libs/app.bundle.min.js?v=6943:178:3210740)
    at https://meet.domain.com/libs/app.bundle.min.js?v=6943:178:3216107
   at https://meet.domain.com/libs/app.bundle.min.js?v=6943:178:3215414
    at Object.next (https://meet.domain.com/libs/app.bundle.min.js?v=6943:178:3215519)
    at o (https://meet.domain.com/libs/app.bundle.min.js?v=6943:178:3214233)
```

### Server information:

- Jitsi Meet version: Tested against 8044, 8218, 8252.

### Client information:

- Browser / app version: Tested using latest Brave / Chrome / Opera

### Additional information:

The bug can be replicated even if you give browser access to camera but not microphone.

When local recording stared, it appears to work, but console log shows the following error:

```
olm.js:11 2023-02-01T20:21:28.738Z [features/base/tracks] Failed to create local tracks ['audio'] 
Gt {gum: {…}, name: 'gum.permission_denied', message: 'User denied permission to use
device(s): audio', stack: 'Error\n    at new Gt (https://meet.domain.com/libs/lib-jitsi-meet.min.js? 
v=6943:2:588990'}
i @ olm.js:11
(anonymous) @ functions.web.ts:140
Promise.catch (async)
(anonymous) @ functions.web.ts:140
Promise.then (async)
jP @ functions.web.ts:106
muteAudio @ conference.js:1064
(anonymous) @ LocalRecordingManager.web.ts:190
(anonymous) @ LocalRecordingManager.web.ts:7
w0 @ LocalRecordingManager.web.ts:3
startLocalRecording @ LocalRecordingManager.web.ts:145
```

### PR DESCRIPTION
**Pull request by robertpin**
Fixes #12857

### CODE REVIEW

File: react/features/recording/components/Recording/LocalRecordingManager.web.ts [hunk 1]
```diff
@@ -231,6 +231,11 @@ const LocalRecordingManager: ILocalRecordingManager = {
                 });
             }
 
+            // handle no mic permission
+            if (!getLocalTrack(getTrackState(getState()), MEDIA_TYPE.AUDIO)?.jitsiTrack?.track) {
```
Comment (saghul) [inline react/features/recording/components/Recording/LocalRecordingManager.web.ts:235]: Why can't we work without that track?

**Review by saghul — APPROVED** [review]
LGTM with a question.

File: react/features/recording/components/Recording/LocalRecordingManager.web.ts (hunk 1 above)
Comment (saghul) [reply react/features/recording/components/Recording/LocalRecordingManager.web.ts:235]: Oh FML. Can we create a fake track with the WebAudio API playing silence perhaps? The current behavior is quite unexpected...

Comment (robertpin) [reply react/features/recording/components/Recording/LocalRecordingManager.web.ts:235]: Hmm, probably. Not sure tho. We'd need some time for this

**Pull request merged**